### PR TITLE
Add retries in verification script and provisioning tests

### PR DESCRIPTION
**Pull request by maelk**
Avoid failing without retrying to wait for the dev env
to initialize properly. Add verify target in Makefile all
Add a provisioning test script that verifies the provisioning and deprovisioning of the machines

### CODE REVIEW

**Comment by nordixinfra** [comment]
Can one of the admins verify this patch?

File: 04_verify.sh [hunk 1]
```diff
@@ -41,67 +76,94 @@ differs(){
...
-      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.spec.bootMACAddress')" \
-        "${mac}" "${name} Baremetalhost mac address correct")"
-
-      # Verify BM host status
-      FAILS="$(equals "$(echo "${BM_HOST}" | jq -r '.status.operationalStatus')" \
-        "OK" "${name} Baremetalhost status OK")"
-
-      # Verify credentials exist
-      CRED_NAME="$(echo "${BM_HOST}" | jq -r '.spec.bmc.credentialsName')"
-      CRED_SECRET="$(kubectl get secret "${CRED_NAME}" -n metal3 -o json | jq '.data')"
```
Comment (namnx228) [inline 04_verify.sh:73]: in the case we use kinder, I think we may need to export KUBECONFIG
```sh
export KUBECONFIG="kinder get kubeconfig-path"
```

Comment (maelk) [reply 04_verify.sh:73]: ok, I'll add a Kubeconfig variable

**Review by kanwar-saad — DISMISSED** [review]

File: 05_test.sh [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+#
+get_vm_ip(){
+  local BMH_NAME="${1}"
+  sudo virsh net-dhcp-leases baremetal | grep "${BMH_NAME}" | awk '{print $5}' \
+    | cut -d '/' -f1
+}
+
+
+# provision the machines
+for name in centos-0 centos-1; do
```
Comment (kanwar-saad) [inline 05_test.sh]: you can pre define this array .. it is used in all loops later

File: 05_test.sh [hunk 3]
```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env bash
```
Comment (kanwar-saad) [inline 05_test.sh:1]: since it is a new file .. you can keep it free of linter errors.

File: lib/common.sh [hunk 4]
```diff
@@ -124,3 +132,88 @@ function list_nodes() {
            .user + " " + .password + " " + .mac' \
        | sed 's/"//g'
 }
+
+iterate(){
```
Comment (kanwar-saad) [inline lib/common.sh:144]: missing function doc ... document description, inputs and outputs.

File: lib/common.sh [hunk 5]
```diff
@@ -124,3 +132,88 @@ function list_nodes() {
...
+    echo -e "======\nFailed\n======\n" >&3
+  elif [[ "${RUNS}" != "0" ]]; then
+    echo -e "=========\nSucceeded\n=========\n" >&3
+  fi
+  echo "${TMP_FAILS}"
+  return "${TMP_RET}"
+}
+
+
+# Check the return code and log
```
Comment (kanwar-saad) [inline lib/common.sh]: same for rest of the functions

**Pull request merged**